src/v/cloud_storage/materialized_segments.h

```diff
@@ -94,25 +119,11 @@ class materialized_segments {
     /// once: this will trigger faster trimming under pressure.
     adjustable_semaphore _segment_units;
 
-    /// Consume from _eviction_list
-    ss::future<> run_eviction_loop();
-
-    /// Try to evict readers until `target_free` units are available in
-    /// _reader_units, i.e. available for new readers to be created.
-    void trim_readers(size_t target_free);
+    /// Size of the materialized_manifest_cache
+    config::binding<size_t> _manifest_meta_size;
 
-    /// Synchronous scan of segments for eviction, reads+modifies _materialized
-    /// and writes victims to _eviction_list
-    void trim_segments(std::optional<size_t>);
-
-    // List of segments to offload, accumulated during trim_segments
-    using offload_list_t
-      = std::vector<std::pair<remote_partition*, model::offset>>;
-
-    void maybe_trim_segment(materialized_segment_state&, offload_list_t&);
-
-    // Permit probe to query object counts
-    friend class remote_probe;
+    /// Cache used to store materialized spillover manifests
+    ss::shared_ptr<materialized_manifest_cache> _manifest_cache;
```

Comment:
nit: could we use a `unique_ptr` here since `get_materialized_manifest_cache` returns a ref. Or did you mean to return `shared_ptr` in `get_materialized_manifest_cache`?